## SUMMARY STATISTICS ASSIGNMENT SOLUTION 

**1. First thing first we to import the packages we need for the process and we also need to create the data given in the problem in to our jupyter notebook in the form of numpy array.**

In [9]:
import numpy as np
import pandas as pd
import statistics
%matplotlib inline

In [3]:
data1 = np.array([6, 8, 8, 10, 11, 12, 14])

In [4]:
data1.mean()

9.857142857142858

In [6]:
np.median(data1)

10.0

In [10]:
statistics.mode(data1)

8

**Therefore we can see from the above results the values of the 3 central tendencies are 9.86(Mean), 10.0(Median) and
8(Mode).**

*The next step is finding the Variance, Standard deviation, and Standard error.*

In [11]:
#the variance of the data
data1.var()

6.408163265306122

In [17]:
#The standard deviation of the data is assigned to a vaiable name std_dev
std_dev = data1.std()

In [18]:
std_dev

2.531435020952764

In [19]:
# Standard error can be calculated as follows
std_error = std_dev/len(data1)**0.5

In [20]:
std_error

0.9567925036515135

**2. If i had to chose only one measure of central tendency, i would chose mean.The reason is that, mean is 
calculated from all the values in the data and the data is not ordinal but rather nominal. And when it comes 
to the measure of variability in the data,i would have selected standard deviation over the other measures especially
when measuring the average indivisual variability. Both the other variation measures have important role to play 
in the analysis.Variation is important to spot outliers at a glance. And standard error is a good measure of 
indivisual samples variability average.**

**3. updating cindy's age to 7 and running the estimates;**


In [22]:
#generated the new data with updates cindy's age
data2 = np.array([7, 8, 8, 10, 11, 12, 14])
#the mean
data2.mean()

10.0

In [23]:
#The Median
np.median(data2)

10.0

In [24]:
#the Mode
statistics.mode(data2)

8

**From the measures of central tendency only ** mean ** is changed after updating one value in the data.**

In [26]:
#moving forward with variability measures
#Variance
data2.var()

5.428571428571429

In [27]:
#standard Deviation
data2.std()

2.32992949004287

In [29]:
#Standard Error
data2.std()/len(data2)**0.5

0.8806305718527109

**There was slight difference in the 3 of the variability measures as all of them involve the entire dataset in 
the calculation as the mean in measure of central tendency.**

**4. Recalculating the value by replacing 8 years old cousine Oliver by 1 year old jessica,will alter the data.**

In [30]:
#lets call the new data data3
data3 = np.array([1, 6, 7, 8, 10, 11, 12, 14])

In [31]:
data3.mean()

8.625

In [32]:
np.median(data3)

9.0

In [33]:
statistics.mode(data3)
#There is no mode!!

StatisticsError: no unique mode; found 8 equally common values

**From this new data,i would prefer "Median" as the best measure of central tendency.The reason is the**
**issue of an outlier(the skewness) in our data.Our data does not have a *mode*.**

**Continuing with the measure of variation;**

In [34]:
#variance
data3.var()

14.484375

In [35]:
#standard deviation
data3.std()

3.8058343369095824

In [36]:
#standard error
data3.std()/len(data3)**0.5

1.3455656338506865

**almost always it is good to calculate variance,because it is a great indicator of the presence of outliers.and 
also we can get both standard deviation and standard error from variance;it is a wonderful measure of the overall 
variability in a dataset.**

**5.From the given dataset to have an understanding about the nature of the data,i decided to check on variance first.**


In [53]:
#First step is generating the dataset in to numpy array or DataFrame
data = np.array([5, 17, 20, 23])
data.var()

46.6875

**From the value of variance i have statistically calculated and concluded there is an outlier.So i chose 
median as the measure of central tendency.** 

In [54]:
#Calculating median
np.median(data)

18.5

**Therefore an estimated 18.5 adult americans Brady Bunch fans were on the 50th anniversary of the show.**